plugins/numbers/numbers.cpp

```diff
@@ -49,14 +49,14 @@ QWidget* Numbers::createEditor(QWidget * parent, const QModelIndex & index)
 	Q_UNUSED(index);
 	if (value().type() == QVariant::Double)
 	{
-		QDoubleSpinBox *dsb = new QDoubleSpinBox(parent);
-		dsb->setMaximum(1e10);
-		dsb->setMinimum(-1e10);
-      dsb->setDecimals(6);
-		dsb->setSingleStep(0.1);
-		dsb->setKeyboardTracking(false);
-		connect(dsb, SIGNAL(valueChanged(double)), this, SLOT(setValue(double)));
-		return dsb;
+		QLineEdit *le = new QLineEdit(parent);
+        connect(le, &QLineEdit::textChanged, this,
+			[this](QString value)
+			{
+				setValue(value.toDouble());
```

Comment:
When value cannot be parsed correctly to double, it returns 0.0, which may be invalid for certain use-cases. In case of a spin box, we could set minimum and maximum range.